pymtl3/passes/tracing/CLLineTracePass.py

```diff
@@ -12,10 +12,24 @@
 
 class CLLineTracePass( BasePass ):
 
+  # CLLineTracePass public pass data
+
+  #: enable
+  #:
+  #: Type: ``bool``; input
+  #:
+  #: Default value: True
+  enable = MetadataKey(bool)
+
   def __init__( self, default_trace_len=8 ):
     self.default_trace_len = default_trace_len
 
   def __call__( self, top ):
+
+    # Turn on by default
+    if top.has_metadata( self.enable ) and top.get_metadata( self.enable ) is False:
+      return
+
     if not hasattr( top, "_tracing" ):
```

Comment:
Similar here. Do we want to leave this as is for 3.0? I mean in general I don't think exposing internal data as an attribute of the component is a good idea. That makes sense only in cases where the pass users are meant to use the added attributes (like `sim_tick()`).

Reply:
I fixed most of them. For simulation-related passes, we can fix them later